SPARCbot.py

```diff
@@ -168,8 +184,276 @@ async def iwantto(self, ctx, role: str):
+                return
+
+            # update bet status
+            if bet_log[bet_key]['status'] in ('open','standing'):
+
+                # taking a standing bet creates a new bet
+                if bet_log[bet_key]['status'] == 'standing':
+                    bet_log = self.add_new_bet(bet_log[bet_key]['bidder'],bet_log[bet_key]['statement'],'standing')
+
+                bet_log[bet_key]['seller'] = ctx.author.id
+                bet_log[bet_key]['status'] = 'pending'
+
+                #save log
+                with open("bet_log.json","w") as write_file:
+                    json.dump(bet_log,write_file,indent=4)
+
+                nick = await get_nick_from_id(ctx,ctx.author.id)
+                await timed_send(ctx,'Bet '+str(bet_id)+' has been claimed by '+nick+'!')
+
+            #bet has already been claimed
+            else:
+                await timed_send(ctx,'That bet\'s not up for grabs!')
+        except KeyError as e:
+            await timed_send(ctx,'That bet doesn\'t exist!')
+            raise
+        except Exception:
+            await timed_send(ctx, 'Hmm...that didn\'t seem to work.')
+            raise
+
+    @commands.command()
+    async def resolve(self,ctx,bet_id):
+        '''<bet_id>: resolve a pending bet or delete an open bet by id'''
+        # TODO: add information to trigger the econ-bot depending on who won
+        with open("bet_log.json", "r") as read_file:
+            bet_log = json.load(read_file)
+        bet_key = 'bet_'+str(bet_id)
+        try:
+            # only participants can resolve their own bets
+            if self.check_author(ctx,bet_id):
+
+                #open bids can be annulled
+                if bet_log[bet_key]['status'] in ('open','standing'):
+                    removed = bet_log.pop(bet_key)
+                    with open("bet_log.json","w") as write_file:
+                        json.dump(bet_log,write_file,indent=4)
+                    nick = await get_nick_from_id(ctx,ctx.author.id)
+                    await timed_send(ctx,'removed bet '+str(bet_id)+' by '+nick)
+
+                #pending bids get set to resolved
+                elif config['bet_status'][bet_log[bet_key]['status']] == 'pending':
+                    bet_log[bet_key]['status'] = 'resolved'
+
+                    with open("bet_log.json","w") as write_file:
+                        json.dump(bet_log,write_file,indent=4)
+
+                    await timed_send(ctx,'Bet '+str(bet_id)+' has been resolved. Use !give-money <@winner> <value> to settle up.')
+            else:
+                await timed_send(ctx,'You can\'t resolve bets you aren\'t a part of!')
+        except KeyError as e:
+            await timed_send(ctx,'That bet doesn\'t exist!')
+            raise
+        except Exception:
+            await timed_send(ctx, 'Hmm...that didn\'t seem to work.')
+            raise
+
+    @commands.command()
+    @commands.has_role('student')
```

Comment:
is this correct? should be 'admin'? also this should be a config parameter because the role names might be different